### ISSUE TITLE
NullPointerException when Epoxy model is bind to data class

### ISSUE THREAD

**Issue by eric-ampire** [airbnb/epoxy#1190]
I have noticed that epoxy throws this exception when the model use the data class with kotlin 1.5.10 and 1.5.0

```
java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.hashCode()' on a null object reference
        at org.zxconnect.android.daylivro.models.entity.Product.hashCode(Unknown Source:19)
        at org.zxconnect.android.delivro.ProductBindingModel_.hashCode(ProductBindingModel_.java:330)
```

**Comment by elihart** [comment]
sounds like a bug in your `Product` class's hashCode implementation calling hashCode on a null string

**Comment by eric-ampire** [comment]
Yes, it is the exception that is triggered, but if I remove when I remove the keyword data, there is no more exception

**Comment by tekinumut** [comment]
I had a non-nullable String variable in my data class but when my Api service started to send null variable to it, this error occurred. 

Downgrade Kotlin version to 1.4.32 is also works but it's not a good option.

So i fixed this issue by converting some of my non-nullable variables to nullable variables like below.

Before:
```
data class ExampleClass {
      val myNullableString: String
      ..
}
```

After:
```
data class ExampleClass {
      val myNullableString: String?=null
      ..
}
```

**Comment by zhambylgaziz** [comment]
Hello, got the same error. I had data class inside first data class, so when this is null, it throws NPE for hashCode(). Appeared after updating Kotlin version to 1.5.0

![image](https://user-images.githubusercontent.com/37974317/152475877-41652f18-5e9c-49fe-9b68-954a60e723a3.png)

**Comment by tekinumut** [comment]
@zhambylgaziz i guess you must initialize your data class variable as nullable like me.

**Comment by wh173d3v11** [comment]
@tekinumut Thanks, i stuck with same issue, your reply solved that :)
